engine/runtime/src/types/rand.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::*;
+
+use libsodium_sys::randombytes_buf;
```

Comment:
is this better than crypto::random ?

Reply:
For now, I'd rather not bring in the crypto.rs dependency just for the random call.  Also, crypto.rs doesn't allow you to specify the length and the pointer at call.  Where as for libsodium, you have more control.